include/irr/asset/CTexturePacker.h

```diff
@@ -43,13 +104,13 @@ class CTexturePacker
         auto& region = regions->front();
         region.imageSubresource.mipLevel = 0u;
         region.imageSubresource.baseArrayLayer = 0u;
-        region.imageSubresource.layerCount = 1u;
+        region.imageSubresource.layerCount = _numLayers;
         region.bufferOffset = 0u;
-        region.bufferRowLength = _szxy;
+        region.bufferRowLength = megaimgSz;
         region.bufferImageHeight = 0u; //tightly packed
         region.imageOffset = {0u, 0u, 0u};
-        region.imageExtent = {_szxy,_szxy,1u};
-        auto buffer = core::make_smart_refctd_ptr<ICPUBuffer>(getTexelOrBlockBytesize(_format)*_szxy*_szxy);
+        region.imageExtent = {megaimgSz,megaimgSz,1u};
+        auto buffer = core::make_smart_refctd_ptr<ICPUBuffer>(getTexelOrBlockBytesize(_format)*megaimgSz*megaimgSz*_numLayers);
         m_megaimg->setBufferAndRegions(std::move(buffer), regions);
     }
 
```

Comment:
pack an ICPUImage subresource, the struct that ICPUImageView uses for its creation (so we don't always have to pack the entire image)